### PR TITLE
Initial specifications for center.bllFlow()

### PR DESCRIPTION
**Pull request by DougManuel**
Please comment if this bllflow centering function makes sense. 
1) It is a bit different from what we discussed.
2)  I realize that we need to define the different centering methods in the MSW. 
3) I don't think we need to have "median" at this point, but should be straightforward to add.

The only files that were added modified where:
- center.Rmd
- PBC - variables.csv (updated center variable)

### CODE REVIEW

File: vignettes/center.Rmd [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+center.bllFlow() # will call sj::center() with the following attributes
+
+#  sj::center  <- function(x, ..., include.fac = FALSE, append = TRUE, suffix = "_C")
+#    where
+#  x = bllFlow data frame
+#  ... = MSW variables where there is an allowable `center` value
+#  note: suffix for bllFlow = "_C" (not "_c")
+#  note: we would create a bllflow method for mean() that would calculate the three different means: 'mean' the original mean; 'meanWeighted' the weighted mean (see weighted.mean {stats}) and 'MSWMean' the mean calcualted using the MSW mean values.
+
+# Do we want a little utility function addMeanToMSW which would calculate mean and then add to the MSW sheet (object)?
```
Comment (DarkShadeKnigh) [inline vignettes/center.Rmd]: I believe we should add to MSW by default to make sure the new renamed variables are part of the variables MSW. Unless we wanna avoid renaming variables and just storing the history of changes performed

File: R/center.R [hunk 2]
```diff
@@ -10,13 +10,13 @@
 #' @examples
 center.bll_flow <- function(x, mean) {
   # Create the centered object
-  centered_x <- x - mean;
-  
+  centered_x <- x - mean
+
   # Make it an instance of the bll_flow class
-  bll_flow(centered_x, cat(attr(x, 'label'), 'Centered'));
+  bll_flow(centered_x, cat(attr(x, "label"), "Centered"))
```
Comment (DougManuel) [inline R/center.R]: @DarkShadeKnigh see some of this old code for ideas. Worthwhile to describe in this PR how you'll implement centered attributes and labels.

File: R/centered-variable.R [hunk 3]
```diff
@@ -9,8 +9,8 @@ library(glue)
 #'
 #' @examples
 getCodeForCenteredVariable <- function(variableNameToCenter) {
-  centeredVariableName <- glue::glue('{variableNameToCenter}_c')
-  
+  centeredVariableName <- glue::glue("{variableNameToCenter}_c")
+
   # The final code looks like:
```
Comment (DougManuel) [inline R/centered-variable.R]: This is a nice description that could be included in our new code.

File: R/centered-variable.R [hunk 4]
```diff
@@ -20,4 +20,4 @@ getCodeForCenteredVariable <- function(variableNameToCenter) {
   # 3. Name matches with the name of the centeredVariableName var and Means
   #    has the mean for it
   return(glue::glue('{centeredVariableName} <- bllFlow::center({variableNameToCenter}, means[means$Name=="{centeredVariableName}", ]$Means)'))
```
Comment (DougManuel) [inline R/centered-variable.R]: @yulric How do you want to hand how we'll generate the pmml converstion. Should @DarkShadeKnigh create R code like you've done previously. (maybe as an option?)

File: vignettes/center.Rmd [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+data(pbc)
+```
+
+```{r}
+library(bllFlow)
+variablesSheet <- read.csv(file.path(getwd(), '../inst/extdata/PBC/PBC - variables.csv'))
+variableDetailsSheet <- read.csv(file.path(getwd(), '../inst/extdata/PBC/PBC - variable_details.csv'))
+pbcModel <- BLLFlow(pbc, variablesSheet, variableDetailsSheet)
+```
+center.bllFlow() centers variables in the bllFlow dataframe using center specifications in the MSW.
```
Comment (DougManuel) [inline vignettes/center.Rmd]: Update name to `center` then have BLLFlow instance examine class, as discussed.

File: vignettes/center.Rmd (hunk 1 above)
Comment (DougManuel) [reply vignettes/center.Rmd]: 1) We'll definitely want to have the provision to create a codebook and/or update an instance of MSW. 
2) As discussed, let's default to renaming variables for transformations like `center` that change every row in a data frame.
3) suggest that we have by default all of the following: rename variable, add `center` attribute, append variable label to include " (centered).
We may want to include how we centered in the attribute. I'm not sure if it should be something like:
`attr(centered_x, 'center') <- TRUE;` and `attr(centered_x, 'centerOn') <- 'mean';
or
`attr(centered_x, 'center') <- 'mean';

**Pull request closed without merging**